File: frontend/src/types/factors.ts
export interface FactorTopPoolItem {
  symbol: string
  name: string
  industry: string
  rank: number
  composite_score: number
  factor_values: Record<string, number | null>
}

export interface FactorTopPoolResponse {
  source: string
  data_date: string | null
  refreshed_at: string | null
  calculation_scope: string
  items: FactorTopPoolItem[]
  total: number
}

export interface FactorRefreshTaskResponse {
  task_id: string
  status: 'pending' | 'running' | 'succeeded' | 'failed'
  created_at: string
  finished_at: string | null
  error_message: string | null
}

export interface FactorOverviewItem {
  factor: string
  name: string
  category: string
  direction: number
  direction_text: string
  weight: number
}

export interface FactorOverviewResponse {
  source: string
  items: FactorOverviewItem[]
  total: number
}

File: backend/app/schemas/factors.py
from pydantic import BaseModel, Field


class FactorTopPoolItem(BaseModel):
    symbol: str = Field(description="六位股票代码")
    name: str
    industry: str
    rank: int = Field(ge=1)
    composite_score: float
    factor_values: dict[str, float | None]


class FactorTopPoolResponse(BaseModel):
    source: str
    items: list[FactorTopPoolItem]
    total: int = Field(ge=0)


class FactorOverviewItem(BaseModel):
    factor: str
    name: str
    category: str
    direction: int
    direction_text: str
    weight: float
    data_source: str
    description: str
    spearman_ic_adjusted: float | None = None
    pearson_ic_adjusted: float | None = None
    ic_positive_rate: float | None = None
    group1_minus_group5: float | None = None
    group_spread_positive_rate: float | None = None
    validation_n: float | None = None
    snapshot_count: int | None = None


class FactorOverviewResponse(BaseModel):
    source: str
    items: list[FactorOverviewItem]
    total: int = Field(ge=0)
